misc/architecture-model-generator/src/main/java/io/ballerina/architecturemodelgenerator/generators/entity/EntityModelGenerator.java

```diff
@@ -110,36 +144,48 @@ private Attribute getAttribute(RecordFieldSymbol recordFieldSymbol, String entit
                     CardinalityValue.ONE_AND_ONLY_ONE.getValue(), associateCardinality));
             associations = new LinkedList<>(List.of(association));
         } else if (fieldTypeDescKind.equals(TypeDescKind.ARRAY) &&
-                ((ArrayTypeSymbol) fieldTypeSymbol).memberTypeDescriptor().typeKind().equals(TypeDescKind.RECORD)) {
-            RecordTypeSymbol inlineRecordTypeSymbol = (RecordTypeSymbol) ((ArrayTypeSymbol)
-                    fieldTypeSymbol).memberTypeDescriptor();
-            fieldType = TypeDescKind.RECORD.getName() + ARRAY;
-            String associateCardinality = optional ? CardinalityValue.ZERO_OR_MANY.getValue() :
-                    CardinalityValue.ONE_OR_MANY.getValue();
-            Association association = new Association(inlineRecordName, new Association.Cardinality(
-                    CardinalityValue.ONE_AND_ONLY_ONE.getValue(), associateCardinality));
-            associations = new LinkedList<>(List.of(association));
-            this.types.put(inlineRecordName, getType(inlineRecordTypeSymbol, inlineRecordName,
+                getInlineRecordTypeSymbol((ArrayTypeSymbol) fieldTypeSymbol) != null) {
+            RecordTypeSymbol inlineRecordTypeSymbol = getInlineRecordTypeSymbol((ArrayTypeSymbol) fieldTypeSymbol);
+            int arraySize = ((ArrayTypeDescriptorNode) recordFieldNode.typeName()).dimensions().size();
+            fieldType = TypeDescKind.RECORD.getName() + ARRAY.repeat(arraySize);
+            RecordTypeDescriptorNode recordTypeDescNode =
+                    (RecordTypeDescriptorNode) ((ArrayTypeDescriptorNode) recordFieldNode.typeName()).memberTypeDesc();
+            this.types.put(inlineRecordName, getType(inlineRecordTypeSymbol, recordTypeDescNode, inlineRecordName,
                     getElementLocation(recordFieldSymbol), true));
-
+            List<String> associateCardinalities = getArrayAssociateCardinalities(recordFieldNode);
+            List<Association> associationsTemp = new LinkedList<>();
+            associateCardinalities.forEach(associationCardinality -> {
+                Association association = new Association(inlineRecordName, new Association.Cardinality(
+                        CardinalityValue.ONE_AND_ONLY_ONE.getValue(), associationCardinality));
+                associationsTemp.add(association);
+            });
+            associations = associationsTemp;
         } else {
-            associations =
-                    getAssociations(recordFieldSymbol.typeDescriptor(), entityName, optional, nillable);
-
+            associations = getAssociations(fieldTypeSymbol, recordFieldNode, entityName, optional, nillable);
         }
         // todo: address when union types has anonymous records
         return new Attribute(fieldName, fieldType, optional, nillable, defaultValue, associations,
                 getElementLocation(recordFieldSymbol));
     }
 
+    private RecordTypeSymbol getInlineRecordTypeSymbol(ArrayTypeSymbol arrayTypeSymbol) {
```

Comment:
Shall we make this return an Optional `RecordTypeSymbol`, instead of returning null?